src/app/src/main/java/com/cmput301f21t09/budgetprojectname/AuthorizationService.java

```diff
@@ -1,5 +1,82 @@
 package com.cmput301f21t09.budgetprojectname;
 
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.auth.AuthResult;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+/**
+ * Authorization service. Handles user sign in, sign out, and registration.
+ */
 public class AuthorizationService {
+    /**
+     * Service
+     */
+    private static AuthorizationService service;
+
+    /**
+     * Firebase auth instance
+     */
+    private final FirebaseAuth firebaseAuth;
+
+    /**
+     * Constructor. Configures and instantiates authorization service.
+     */
+    private AuthorizationService() {
+        firebaseAuth = FirebaseAuth.getInstance();
+    }
+
+    /**
+     * Get authorization service
+     * @return Authorization service
+     */
+    public static AuthorizationService getInstance() {
+        if (service == null) {
+            service = new AuthorizationService();
+        }
+
+        return service;
+    }
+
+    /**
+     * Sign in with given email and password
+     * @param email The email to sign in with
+     * @param password The password to sign in with
+     * @return `Task` of `AuthResult` with the result of the sign in attempt
+     */
+    public Task<AuthResult> signIn(String email, String password) {
+        return firebaseAuth.signInWithEmailAndPassword(email, password);
+    }
+
+    /**
+     * Register with given email and password
+     * @param email The email to register
+     * @param password The password to register
+     * @return `Task` of `AuthResult` with the result of the registration attempt
+     */
+    public Task<AuthResult> register(String email, String password) {
+        return firebaseAuth.createUserWithEmailAndPassword(email, password);
+    }
+
+    /**
+     * Sign out
+     */
+    public void signOut() {
+        firebaseAuth.signOut();
+    }
+
+    /**
+     * Check if signed in
+     */
+    public boolean isSignedIn() {
+        return firebaseAuth.getCurrentUser() != null;
+    }
 
+    /**
+     * Get current signed in user
+     * @return The current signed in user. Return null if no user is signed in
+     */
+    public FirebaseUser getCurrentUser() {
```

Comment:
Instead of returning the FirebaseUser object, can we just return the user's UUID?

Reply:
Don't want any components to need to worry about the specific backend we are using

Reply:
Yep, I can definitely do that